src/bookshelf/index.ts

```diff
@@ -25,38 +25,40 @@ export default class Bookshelf implements Mapper {
    * The `any` type data source is set for typing compatibility, but must be removed if possible
    * TODO fix data any type
    */
-  map(data: Data | any, type: string, bookOpts: BookOpts = {}): any {
-
-    let linkOpts: LinkOpts = { baseUrl: this.baseUrl, type, pag: bookOpts.pagination };
+  map(data: Data | any, type: string, mapOpts: MapOpts = {}): any {
```

Comment:
I made some `BookOpts` properties _non-nullable_ because they are set with default values if the user doesn't pass them, but the options object should let them be `null`/`undefined`. So we receive them as `MapOpts` (all nullable) and use them as `BookOpts` most _non-nullable_.